src/main/java/org/jitsi/jigasi/transcription/OracleTranscriptionService.java

```diff
@@ -0,0 +1,350 @@
+    {
+        websocketUrlConfig = JigasiBundleActivator.getConfigurationService()
+                .getString(WEBSOCKET_URL, DEFAULT_WEBSOCKET_URL);
+        compartmentId = JigasiBundleActivator.getConfigurationService()
+                .getString(COMPARTMENT_ID, "");
+    }
+
+    public boolean supportsLanguageRouting()
+    {
+        return false;
+    }
+
+
+    @Override
+    public boolean supportsStreamRecognition()
+    {
+        return true;
+    }
+
+    @Override
+    public boolean supportsFragmentTranscription()
+    {
+        return false;
+    }
+
+    @Override
+    public void sendSingleRequest(final TranscriptionRequest request,
+                                  final Consumer<TranscriptionResult> resultConsumer)
+    {
+        logger.warn("The OCI transcription service does not support single requests.");
+    }
+
+    @Override
+    public StreamingRecognitionSession initStreamingSession(
+            Participant participant)
+    {
+        return new OracleStreamingSession(websocketUrlConfig);
+    }
+
+    @Override
+    public boolean isConfiguredProperly()
+    {
+        return true;
+    }
+
+    public class OracleStreamingSession implements StreamingRecognitionSession, OracleRealtimeClientListener
+    {
+        private OracleRealtimeClient client;
+
+        private UUID uuid = UUID.randomUUID();
+
+        private boolean sessionEnding = false;
+
+        private final String websocketUrl;
+
+
+        /**
+         * List of TranscriptionListeners which will be notified when a
+         * result comes in
+         */
+        private final List<TranscriptionListener> listeners = new ArrayList<>();
+
+        private Instant sessionStart;
+
+
+        public OracleStreamingSession(String websocketUrl)
+        {
+            this.websocketUrl = websocketUrl;
+            try
+            {
+                client = new OracleRealtimeClient(
+                        this,
+                        authProvider,
+                        compartmentId);
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while creating OCI client", e);
+            }
+        }
+
+        private void connect()
+        {
+            if (client.isConnected())
+            {
+                return;
+            }
+
+            final RealtimeParameters realtimeClientParameters = RealtimeParameters.builder()
+                    .isAckEnabled(false)
+                    .finalSilenceThresholdInMs(1000)
+                    .build();
+            try
+            {
+                client.open(websocketUrl, 443, realtimeClientParameters);
+                sessionStart = Instant.now();
+            }
+            catch (Exception e)
+            {
+                logger.error("Error while connecting to OCI service", e);
+            }
+        }
+
+        @Override
+        public void sendRequest(TranscriptionRequest request)
+        {
+            connect();
```

Comment:
Is this the reconnection logic? If the connection fails it will be marked as not connected and it will establish a new connection, right?

Reply:
Exactly. It is needed because Oracle will disconnect you after a few seconds of inactivity (not sending audio).

Reply:
This calls `client.open()` every time until `client.isConnected()`. I think it's used on every audio packet, so we can end up calling `open` many times before the WS is actually connected.